modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -1469,6 +1504,26 @@ private static Iterable<ClusterNode> nodesPerJvm(Iterable<ClusterNode> nodes) {
         return grpMap.values();
     }
 
+    /**
+     * Returns one representative node for each host.
+     *
+     * @param nodes Nodes.
+     * @return Collection which contains only one representative node for each host.
+     */
+    private static Iterable<ClusterNode> nodesPerHost(Iterable<ClusterNode> nodes) {
```

Comment:
It looks like that we already have the "IgniteUtils#neighborhood(nodes)" method, which does a similar thing and which is already used in the GridEntLicenseProcessor.